### DISCUSSION TITLE
noob: pip install of open-webui does not work with PORT set to other than 8080

### DISCUSSION THREAD

**Discussion by richtong** [open-webui/open-webui#6486] [Q&A] [answered]
OK, I'm really new to open-webui and it is awesome!

But one question, with the latest 0.35, it looks like, at least with the pip install open-webui that, the --port no longer works

If I try

```
pip install open-webui
# note that as of 0.35, it also needs tf-keras to run
# this works fine or without the port argument
open-webui serve --port 8080
# but this fails with on when you first try to access localhost:1314
open-webui server --port 1314
# with a not found error
```

Here is the error at the chat/tags/all endpoint

```
INFO  [open_webui.apps.openai.main] get_all_models()
INFO:     127.0.0.1:57628 - "GET /api/v1/knowledge/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:57630 - "GET /api/v1/prompts/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:57647 - "GET /api/v1/chats/tags/all HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:57645 - "GET /api/v1/tools/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:57646 - "GET /api/v1/configs/banners HTTP/1.1" 200 OK
INFO:     127.0.0.1:57628 - "GET /api/v1/functions/ HTTP/1.1" 200 OK
```

What am I missing? I'm sure alot

**Comment by richtong** [comment]
Also does it make sense, I'm happy to do a PR to update the installation documentation as Python 3.11 is required and tf-keras is as well.

The pip install is really useful for quick tests, we can revert to docker if needed, but man is it fast without it :-)

**Reply by richtong** [reply]
Finally for those of you wondering how to debug stuff like this it took me a while, but if you want to reset configurations then you have to dig pretty deep to find the configurations, they do not live in any conventional places like XDG_CONFIG, but are in the pip package storage itself. 

As an example, I use asdf to give me versioned python, so they would live in for a PYTHON_VERSION=3.11.8

```
~/.asdf/installs/python/3.11.8/lib/python3.11/site-packages/open_webui/data/webui.db
```
If you remove the webui.db, then it will reset the whole configuration. This looks like it is the equivalent of the docker data volume.

If you are using a virtual env, then it will be in wherever your python binary is takes you in the ../data directory there

```
cd $(dirname $(command -v python))/../lib/python3.11/site-packages/open_webui/data
```

Which is also pretty far down, the main thing is to look for the site-packages directory

**Reply by richtong** [reply]
YOu can tell if you are there if there is a readme.txt that refers to the storage directory for the backend

**Reply by putrasto** [reply, marked as answer]
I got same error like yours but I clone and run locally using terminal and vscode following this documentation https://docs.openwebui.com/getting-started/development
When I compared the source code in browser and my local source code, it seems they are not the same.
I tried to hard refresh or private mode in browser, it does not solve but after I re-build using "npm run build", the error 
/api/v1/chats/tags/all HTTP/1.1" 404 Not Found, is gone.

**Reply by tne-ops** [reply]
Tht's pretty confusing. Sounds like the pip package is out of date.
